### ISSUE TITLE
arrow-parens incompatible with flowtype

### ISSUE THREAD

**Issue by camjackson** [babel/eslint-plugin-babel#55]
Not sure if this is the right repo for this issue, happy to go complain elsewhere :)

I'm running arrow-parens as-needed, but I'm also using flowtype. eslint complains that the parens are unnecessary on this type-annotated, single-parameter arrow function

```js
const fn = (a: number) => {
  console.log(a);
};
```

I tried removing the parens, but that seems to be invalid syntax, as everything breaks at that point.

**Comment by hzoo** [comment]
This should of been covered by https://github.com/babel/eslint-plugin-babel/pull/44/files?

**Comment by aaron-harvey** [comment]
I'm also facing a similar issue with flowtype annotations on arrow functions.
```javascript
// arrow-parens": [2, "as-needed", { "requireForBlockBody": true }],
const bar = (foo: FooType) => ({});
// "Unexpected parentheses around single function arg..."
``` 
I've had to disable the rule for now.

**Comment by matteocng** [comment]
@aaron-harvey the following .eslintrc configuration works for me with eslint-plugin-babel@3.30
```
...
"rules":
{
"arrow-parens": 0,
"babel/arrow-parens": 1
}
...
```

**Comment by aaron-harvey** [comment]
@matteocng I completely missed the fact it was a default eslint rule vs babel rule.  Thank you!
